### PR TITLE
[ci][build] fix vllm-flash-attn

### PR DESCRIPTION
**Pull request by youkaichao**

### CODE REVIEW

**Review by mgoin — APPROVED** [review]
LGTM but should we just push a placeholder directory into the source?

**Comment by youkaichao** [comment]
I find that it actually works for ci, because ci uses `python setup.py` to build the wheel.

what's wrong is when we use `pip install -e .` , which will miss the python files.

anyway, after this pr, `pip install -e .` should work too.

FYI @ProExpertProg

**Comment by ywang96** [comment]
Man I was wondering why it kept failing build on my dev environment and I thought it's something wrong with my env 🤦

Thanks for fixing!

**Comment by mgoin** [comment]
Again, thank you @youkaichao 🙏

**Comment by youkaichao** [comment]
@ywang96 🤝

File: setup.py [hunk 1]
```diff
@@ -258,6 +258,21 @@ def build_extensions(self) -> None:
...
+        # copy vllm/vllm_flash_attn/*.py from self.build_lib to current
+        # directory so that they can be included in the editable build
+        import glob
+        files = glob.glob(
+            os.path.join(self.build_lib, "vllm", "vllm_flash_attn", "*.py"))
+        for file in files:
+            dst_file = os.path.join("vllm/vllm_flash_attn",
+                                    os.path.basename(file))
+            print(f"Copying {file} to {dst_file}")
+            self.copy_file(file, dst_file)
```
Comment (ProExpertProg) [inline setup.py:274]: Why is this necessary? Is the install step for ``vllm_flash_attn_c` not enough?

Comment (youkaichao) [reply setup.py:274]: see https://github.com/vllm-project/vllm/pull/8699#issuecomment-2365483678
